tests/torchtune/utils/test_generation.py

```diff
@@ -102,6 +116,38 @@ def test_reproducibility(self, generation_model, prompt_tokens):
 
         assert outputs_first == outputs_second
 
+    def test_reproducibility_kv_cache_vs_no_kv_cache(
```

Comment:
Wouldn't it have been more light weight to just take the original forward pass test and extend it to a few more steps?